src/panels/lovelace/components/hui-conditional.ts

```diff
@@ -0,0 +1,154 @@
+  // todo: add row (can't find any createhuirow or LovelaceRowConfig...)
+  // rows: Lovelace;
+  // use it with createRowElement
+}
+
+export class HuiConditional extends LitElement {
+  protected _card?: LovelaceCard;
+
+  private _hass?: HomeAssistant;
+  private _conditionalConfig?: ConditionalConfig;
+  private _elements: LovelaceElement[] = [];
+
+  set hass(hass: HomeAssistant) {
+    this._hass = hass;
+
+    if (this.hasNoItems()) {
+      return;
+    }
+
+    const visible = this.handleItems();
+
+    // This will hide the complete component so it won't get styled by parent in case there's nothing in it
+    this.style.setProperty("display", visible ? "" : "none");
+  }
+
+  public setConditionalConfig(config) {
+    if (
+      !config.conditions ||
+      !Array.isArray(config.conditions) ||
+      !config.conditions.every((c) => c.entity && (c.state || c.state_not)) ||
+      (!config.card && (!config.elements || !Array.isArray(config.elements)))
+    ) {
+      throw new Error("Error in conditional configuration.");
+    }
+
+    this._conditionalConfig = config;
+
+    // set style (overridable)
+    // LitElement static styles can't work since we want to allow deriving classes to modify it
+    const styleElement: HTMLStyleElement = document.createElement("style");
+    styleElement.innerHTML = this.getCustomStyle();
+    this.shadowRoot!.appendChild(styleElement);
+
+    this.removeItems();
+
+    this.createItems();
+
+    if (this._hass) {
+      this.hass = this._hass;
+    }
+  }
+
+  protected getCustomStyle(): string {
+    return "";
+  }
+
+  protected _createHuiElement(
+    _elementConfig: LovelaceElementConfig
+  ): LovelaceElement {
+    return (null as unknown) as LovelaceElement;
+  }
+
+  private createItems() {
+    if (this._conditionalConfig!.card) {
+      this._card = createCardElement(this._conditionalConfig!.card);
+    } else if (
+      this._conditionalConfig!.elements &&
+      this._conditionalConfig!.elements.length > 0
+    ) {
+      this._conditionalConfig!.elements.map(
+        (elementConfig: LovelaceElementConfig) => {
+          this._elements.push(this._createHuiElement(elementConfig));
+        }
+      );
+    }
+    // todo: add rows, etc'....
+  }
+
+  private handleItems(): boolean {
+    const conditionsMet =
+      this._conditionalConfig &&
+      this._conditionalConfig.conditions.every((c) => {
+        if (!(c.entity in this._hass!.states)) {
+          return false;
+        }
+        if (c.state) {
+          return this._hass!.states[c.entity].state === c.state;
+        }
+        return this._hass!.states[c.entity].state !== c.state_not;
+      });
+
+    if (conditionsMet) {
+      this.appendItems();
+    } else {
+      this.removeItems();
+    }
+
+    return conditionsMet as boolean;
+  }
+
+  private appendItems() {
+    if (this._card) {
+      this._card.hass = this._hass;
+      if (!this._card!.parentNode) {
+        this.shadowRoot!.appendChild(this._card);
```

Comment:
The original conditional card was not using a shadow root, neither should this element.